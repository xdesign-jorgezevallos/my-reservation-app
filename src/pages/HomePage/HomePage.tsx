import React from 'react';
import TemplateBase from '../../components/commons/TemplateBase/TemplateBase';

export const HomePage: React.FC = () => {
  return (
    <TemplateBase>
      <div data-testid="home-page">
        <h1 data-testid="title-page">Welcome Home</h1>
        <p>This is home content</p>
      </div>
      <br />
    </TemplateBase>
  );
};
