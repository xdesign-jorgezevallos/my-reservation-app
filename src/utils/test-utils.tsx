
// ---------------------------------
// USING
// ---------------------------------
// "@reduxjs/toolkit": "1.5.0",
// "@testing-library/react": "11.2.5",
// "typescript": "3.7.2"
// "react": "16.13.1"
// "react-redux": "7.2.0",
// "react-router-dom": "5.2.0",

//import { configureStore } from '@reduxjs/toolkit';
import { render as rtlRender } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store'

interface WrapperProps {
    children?: React.ReactNode;
}

function render(ui: any, { route = '/', initialState = {} } = {}) {
    window.history.pushState({}, 'Test page', route);
    
    const Wrapper = ({ children }: WrapperProps) => {
        return (
            <Provider store={store}>
                <BrowserRouter>{children}</BrowserRouter>
            </Provider>
        );
    };

    return rtlRender(ui, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };