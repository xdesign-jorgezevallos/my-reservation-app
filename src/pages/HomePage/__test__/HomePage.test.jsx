import { render, screen } from '../../../utils/test-utils';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HomePage } from '../HomePage';

// const mockedUsedNavigate = jest.fn();

// jest.mock('react-router-dom', () => ({
//   ...jest.requireActual('react-router-dom'),
//   useNavigate: () => mockedUsedNavigate,
// }));

describe('HomePage', () => {
  //   it('should match homePage snapshot', () => {
  //     const homePage = renderer.create(<HomePage />).toJSON();
  //     expect(homePage).toMatchSnapshot();
  //   });

  it('should match homePage snapshot', () => {
    const wrapper = shallow(<HomePage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render HOME page', () => {
    render(<HomePage />);
    const homePageElement = screen.getByTestId('home-page');
    const titleElement = screen.getByTestId('title-page');

    expect(homePageElement).toBeInTheDocument();
    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveTextContent('Welcome Home');
    expect(homePageElement).toContainHTML('<p>This is home content</p>');
  });
});
