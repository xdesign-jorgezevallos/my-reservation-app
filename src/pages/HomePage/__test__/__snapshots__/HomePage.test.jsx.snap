// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match homePage snapshot 1`] = `
<TemplateBase>
  <div
    data-testid="home-page"
  >
    <h1
      data-testid="title-page"
    >
      Welcome Home
    </h1>
    <p>
      This is home content
    </p>
  </div>
  <br />
</TemplateBase>
`;
